src/features/wallet/root/useActivityItem.tsx

```diff
@@ -0,0 +1,303 @@
+      }
+
+      return true
+    },
+    [isSending, address],
+  )
+
+  const formatAmount = (
+    fee: boolean,
+    amount?: Balance<DataCredits | NetworkTokens> | number,
+  ) => {
+    if (!amount) return ''
+
+    if (typeof amount === 'number') {
+      if (amount === 0) return '0'
+      return `${fee ? '-' : '+'}${amount.toLocaleString()}`
+    }
+
+    if (amount?.floatBalance === 0) return amount.toString(8)
+
+    return `${fee ? '-' : '+'}${amount?.toString(8)}`
+  }
+
+  const amount = useCallback(
+    (item: AnyTransaction | PendingTransaction, convertToHNT = false) => {
+      if (item instanceof AddGatewayV1) {
+        if (!convertToHNT) return animalHash(item.gateway)
+
+        return formatAmount(true, feeToHNT(item.fee))
+      }
+      if (item instanceof TransferHotspotV1) {
+        if (!item.fee) return ''
+
+        if (convertToHNT) {
+          return formatAmount(!isSelling(item), feeToHNT(item.amountToSeller))
+        }
+        return formatAmount(!isSelling(item), item.amountToSeller)
+      }
+      if (item instanceof AssertLocationV1 || item instanceof TokenBurnV1) {
+        if (!item.fee) return ''
+
+        if (convertToHNT) {
+          return formatAmount(true, feeToHNT(item.fee))
+        }
+        return formatAmount(isFee(item), item.fee)
+      }
+      if (item instanceof RewardsV1) {
+        return formatAmount(isFee(item), item.totalAmount)
+      }
+      if (item instanceof PaymentV1) {
+        return formatAmount(isFee(item), item.amount)
+      }
+      if (item instanceof PaymentV2) {
+        return formatAmount(isFee(item), item.totalAmount)
+      }
+      const pendingTxn = item as PendingTransaction
+      if (pendingTxn.txn !== undefined) {
+        if (pendingTxn.txn.type === 'add_gateway_v1') {
+          return animalHash((pendingTxn.txn as AddGatewayV1).gateway)
+        }
+        return formatAmount(isFee(item), pendingTxn.txn.fee)
+      }
+      return (
+        (item as AddGatewayV1).fee?.toString(8) ||
+        (item as TokenBurnV1).amount?.toString(8) ||
+        ''
+      )
+    },
+    [feeToHNT, isFee, isSelling],
+  )
+
+  const time = useCallback(
+    (item: AnyTransaction | PendingTransaction, dateFormat?: string) => {
+      const pending = item as PendingTransaction
+      if (pending.status === 'pending') {
+        return t('transations.pending')
+      }
+      const val = fromUnixTime((item as AddGatewayV1).time)
+
+      if (!dateFormat)
+        return formatDistanceToNow(val, {
+          locale: shortLocale,
+          addSuffix: true,
+        })
+      return format(val, dateFormat)
+    },
+    [t],
+  )
+
+  const snapHeight = useCallback(
+    (item: AnyTransaction | PendingTransaction) => {
+      switch (item.type as TxnType) {
+        case 'payment_v1':
+        case 'payment_v2':
+          return isSending(item) ? 509 : 480
+        case 'transfer_hotspot_v1':
+          return 566
```

Comment:
will these heights work across devices? maybe using a percent of height would be more flexible? if these work though then i'm fine with it